idaes/apps/flexibility_analysis/decision_rules/relu_dr.py

```diff
@@ -0,0 +1,160 @@
+keras, keras_available = attempt_import("tensorflow.keras")
+omlt, omlt_available = attempt_import("omlt")
+omlt_nn, _ = attempt_import("omlt.neuralnet")
+omlt_io, _ = attempt_import("omlt.io")
+plt, _ = attempt_import("matplotlib.pyplot")
+
+
+def construct_relu_decision_rule(
+    input_vals: MutableMapping[_GeneralVarData, Sequence[float]],
+    output_vals: MutableMapping[_GeneralVarData, Sequence[float]],
+    config: ReluDRConfig,
+) -> _BlockData:
+    """
+    Construct a neural network-based decision rule with ReLU activation functions
+    from the data provided for the inputs and outputs.
+
+    Parameters
+    ----------
+    input_vals: input_vals: MutableMapping[_GeneralVarData, Sequence[float]]
+        Data for the variables that are inputs to the decision rule
+    output_vals: input_vals: MutableMapping[_GeneralVarData, Sequence[float]]
+        Data for the variables that are outputs to the decision rule
+    config: ReluDRConfig
+        A config object to specify options for the decision rule
+
+    Returns
+    -------
+    res: _BlockData
+        A pyomo model containing the linear decision rule
+    """
+    tf.random.set_seed(config.tensorflow_seed)
+    inputs = list(input_vals.keys())
+    outputs = list(output_vals.keys())
+    n_samples = len(input_vals[inputs[0]])
+
+    config: ReluDRConfig = config()
+    if config.batch_size > n_samples:
+        config.batch_size = n_samples
+
+    training_input = np.empty((n_samples, len(inputs)))
+    for ndx, inp in enumerate(inputs):
+        training_input[:, ndx] = np.array(input_vals[inp], dtype=np.float64)
+
+    training_output = np.empty((n_samples, len(outputs)))
+    for ndx, outp in enumerate(outputs):
+        training_output[:, ndx] = np.array(output_vals[outp], dtype=np.float64)
+
+    if config.scale_inputs:
+        input_mean = training_input.mean(axis=0)
+        input_std = training_input.std(axis=0)
+        for ndx in range(len(inputs)):
+            training_input[:, ndx] = (
+                training_input[:, ndx] - input_mean[ndx]
+            ) / input_std[ndx]
+    else:
+        input_mean = [0] * len(inputs)
+        input_std = [1] * len(inputs)
+
+    if config.scale_outputs:
+        output_mean = training_output.mean(axis=0)
+        output_std = training_output.std(axis=0)
+        for ndx in range(len(outputs)):
+            training_output[:, ndx] = (
+                training_output[:, ndx] - output_mean[ndx]
+            ) / output_std[ndx]
+    else:
+        output_mean = [0] * len(outputs)
+        output_std = [1] * len(outputs)
+
+    nn = keras.Sequential()
+    nn.add(
+        keras.layers.Dense(
+            units=config.n_nodes_per_layer, input_dim=len(inputs), activation="relu"
+        )
+    )
+    for _ in range(config.n_layers - 1):
+        nn.add(keras.layers.Dense(config.n_nodes_per_layer, activation="relu"))
+    nn.add(keras.layers.Dense(len(outputs)))
+    if config.learning_rate is None:
```

Comment:
I assume this is needed as `keras` does not accept `None` as a value for `learning_rate` (i.e. if you provide the argument it must have a meaningful value). An inline comment might help to make sure no-one tries to "fix" this later.